### PR TITLE
Prepare documentation for Toil paper

### LINKED ISSUE #738: Running on Azure
docs/running 'Running on Azure'

### LINKED ISSUE #741: Fix Documentation For Paper
Master issue

### LINKED ISSUE #744: Fix batch System Interface docs
docs/batchSystem "Implementing the batch system interface tutorial"

### LINKED ISSUE #745: Fix the job store interface docs
docs/jobStore.rst

### LINKED ISSUE #752: Enforce venv'ed Sphinx when building docs
Add check to config.py. See Makefile for how to ~~enforce~~ check for a venv.

### LINKED ISSUE #767: Make venv check in conf.py more informative

### PR DESCRIPTION
**Pull request by hannes-ucsc**
Continued from #742

Resolves #741, resolves #752, resolves #767, resolves #738, resolves #744, resolves #745.

### CODE REVIEW

**Comment by jvivian** [comment]
Just noticed that Chapter 9: The Batch System, is not building correctly. 

On further notice... There is a critical error in the abstractBatchSystem class (which is preventing it from building).

**Comment by jvivian** [comment]
lol who wrote this section in Chapter 8, section 1?  "Toil implements lots of optimizations designed to allow it to scale out. **Many are not worth mentioning**, but we detail some of the key ones below that we have found useful."    I'm going to submit a commit to remove it.

**Comment by hannes-ucsc** [comment]
And guys, we don't do periods at the end of commit messages.

A friend of mine wrote this: http://chris.beams.io/posts/git-commit/

Well worth reading on a rainy day ...

File: docs/architecture.rst [hunk 1]
```diff
@@ -52,9 +52,8 @@ These components are described below:
 Optimizations
 -------------
 
-Toil implements lots of optimizations designed to allow it to scale out. 
-Many are not worth mentioning, but we detail some of the key ones below that we
-have found useful.
+Toil implements lots of optimizations designed for scalability.
```
Comment (hannes-ucsc) [inline docs/architecture.rst]: s/lots/buttloads/g

**Comment by hannes-ucsc** [comment]
I'm going to rebase and merge this. Taking off multi-author.

**Pull request merged**